File: LeetCode/Easy/283 - Move Zeroes/283 - Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    const n = nums.length;
    const zeros = [];
    for (let i = 0; i < n; ++i) {
        if (nums[i]) {
            if (zeros.length) {
                const idx = zeros.shift();
                nums[idx] = nums[i];
                nums[i] = 0;
                zeros.push(i);
            }
        }
        else 
            zeros.push(i);
    }
};

File: LeetCode/Easy/283 - Move Zeroes/283 - Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        queue<int> zeros;
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            if (!nums[i]) zeros.push(i);
            else if (!zeros.empty()) {
                swap(nums[i], nums[zeros.front()]);
                zeros.pop();
                zeros.push(i);
            }
        }
    }
};